### PR TITLE
175636844 Social media auth

### PR DESCRIPTION
**Pull request by MChristian00**
#### What does this PR do?
Makes it possible for a user to login into the Barefoot application via Facebook and Google.
#### Description of Task to be completed?
install  required packages for implementing social media auth
add routes for login using Google or Facebook and their corresponding callbacks
 - GET `localhost:3000/api/v1/user/login/google`
 - GET `localhost:3000/api/v1/user/login/facebook`

document routes using swagger
use the Sequelize ORM to save returned profile data in the database
#### How should this be manually tested?
run `git clone https://github.com/atlp-rwanda/tech-strikers-bn-backend.git`
run `cd directory tech-strikers-bn-backend`
run `git checkout ft-social-media-auth-175636844`
run `touch .env`
add the following in the .env file 
```
GOOGLE_CLIENT_ID = "clientID"
GOOGLE_CLIENT_SECRET = "clientSecret"
FACEBOOK_CLIENT_ID = "clientID"
FACEBOOK_CLIENT_SECRET = "clientSecret"
DEV_DATABASE_URL 
```
run `npm install`
run `npm run dev`
finally, open your browser to test using the following urls:
 - for google use `localhost:3000/api/v1/user/login/google`
 - for facebook use `localhost:3000/api/v1/user/login/facebook`

then check the database for the returned profile data
#### Any background context you want to provide?
Since there's no frontend to display returned profile data, it'll be logged to the terminal.
#### What are the relevant pivotal tracker stories?
[#175636844](https://www.pivotaltracker.com/story/show/175636844)

### CODE REVIEW

**Review by Duncanwar — COMMENTED** [review]
Screenshot from 2020-11-23 20-00-09

While checking it manually, I am getting this error.

**Comment by MChristian00** [comment]
> Screenshot from 2020-11-23 20-00-09
> 
> While checking it manually, I am getting this error.

Yeah, that's because I didn't include the secret keys and IDs. They should be in the .env

**Comment by Thiernope** [comment]
Apart from the passport you told us to install, what are those other requisite packages to install? Review the description of the task to be completed. I am getting this error when running the server.
the error

**Comment by Thiernope** [comment]
Again, it would be better if the conflicts are resolved. Thanks for the work done.

File: .eslintrc.json [hunk 1]
```diff
@@ -42,9 +45,9 @@
...
         "require": {
-          "FunctionDeclaration": true,
-          "MethodDefinition": true,
-          "ClassDeclaration": true
+          "FunctionDeclaration": false,
+          "MethodDefinition": false,
+          "ClassDeclaration": false
         }
       }
     ]
```
Comment (Duncanwar) [inline .eslintrc.json]: why are you disabling this rule

File: package.json [hunk 2]
```diff
@@ -1,12 +1,11 @@
 {
   "name": "express-authorshaven",
   "version": "1.0.0",
-  "type": "module",
   "description": "A Social platform for the creative at heart",
   "main": "src/index.js",
   "scripts": {
     "start": "node --experimental-json-modules src/index.js",
-    "test": "nyc --reporter=text mocha --experimental-json-modules --exit",
+    "test": "nyc --reporter=text mocha --experimental-json-modules -r @babel/register --timeout 10000 --exit",
```
Comment (Duncanwar) [inline package.json]: check on Slack channel a new way of writing test script

**Review by Duncanwar — COMMENTED** [review]
It looks nice

File: .eslintrc.json (hunk 1 above)
Comment (MChristian00) [reply .eslintrc.json]: We did it because we thought jsdocs were unnecessary but I guess we'll have to undo that.

**Review by Muhire-Josue — APPROVED** [review]
LGTM

**Pull request merged**